In [1]:
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd
import scipy.stats as t

# Example of confidence interval

### We have:
1. mean = 10
2. sd = 5
3. n = 100

### We need to calculate 99% confidence interval

In [2]:
n = 100
sd = 5
mean = 10

se = sd/np.sqrt(n)
se

0.5

### Standard error is needed to calculate confidence interval, it is the result of dividing standard deviation by number of samples

In [3]:
(mean - 2.58*se, mean+2.58*se)

(8.71, 11.29)

### 2.58 is the z-score corresponding to 99% of data, thus we with the 99% of confidence we can say that mean of general population is lay between 8.71 and 11.29

# Using real-world data

In [4]:
df = pd.read_csv("../data/genetherapy.csv")
dist = df['expr']
mean = dist.mean()
sd = dist.std()
n = len(dist)

### Calculate 95% confidence interval

In [5]:
se = sd/np.sqrt(n)
(mean-se*1.96, mean+se*1.96)

(94.89479982930055, 97.73853350403277)

### With 95% of confidence we can say that mean of gene expression is lay between 94.89 and 97.74 in general population

# Using interval method in `scipy.stats`

In [6]:
t.t.interval(0.95, len(dist)-1, loc=np.mean(dist), scale=t.sem(dist))

(94.86506005016271, 97.76827328317061)